lib/ecl/well_branch_collection.cpp

```diff
@@ -33,8 +35,8 @@
 struct well_branch_collection_struct {
   UTIL_TYPE_ID_DECLARATION;
 
-  vector_type     *  __start_segments;
-  int_vector_type *  index_map;
+  std::vector<const well_segment_type*> __start_segments;
```

Comment:
There is *something* with `const` members in STL containers; please remove the `const`.